### ISSUE TITLE
Problem setting up SSL connection - SSL routines:ssl3_get_record:wrong version

### ISSUE THREAD

**Issue by farynaio** [znc/znc#1781]
When SSL is activated I get `src/Csocket.cpp:664 error:1408F10B:SSL routines:ssl3_get_record:wrong version number`.

The output of `openssl s_client -connect` for znc server:
```
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-CHACHA20-POLY1305
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-CHACHA20-POLY1305
``` 

ZNC config:
```
SSLCiphers = EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:ECDHE+RSA+AES256+GCM+SHA384
SSLProtocols = -SSLv2 +SSLv3 +TLSv1 +TLSv1.1 +TLSv1.2
```

I use ZNC 1.7.2+deb3.

Any ideas?

**Comment by psychon** [comment]
Are you trying to connect to znc or is this a problem when znc is trying to connect to some IRC server? I am just trying to make sure I understand correctly: You are providing the `openssl` output from trying to connect from your computer to znc, but in #1743 you say that you have the same problem as someone who could not connect from znc to an IRC network.

Also, what are you using to try to connect to znc? As in, are the error message from Csocket and the output from `openssl` that you provide above from the same connection attempt?

**Comment by farynaio** [comment]
Sorry for confusion. I try to connect to ZNC by IRC client. I use [Erc](https://www.gnu.org/software/emacs/erc.html), but I tried one other client with similar effect. The ZNC connects to the networks just fine. I thought the problem is SSLCiphers related, that's why I posted output from the openssl client connecting to ZNC server. 

Currently, I don't get `src/Csocket.cpp:664 error:1408F10B:SSL routines:ssl3_get_record:wrong version number` I mentioned before, it turned out I started ZNC with SSL but tried to connect without. Right now I don't get any message on the server side and the client just stays forever on "Opening Connection.." stage.

I put a bit more details in:
https://emacs.stackexchange.com/questions/63639/erc-znc-over-ssl
